### PR TITLE
Refactor the way module documentation is collected for classes

### PR DESCRIPTION
**Pull request by Xrayez**
## Obsolete
`get_doc_classes()` in `config.py` - no need to list individual classes (doesn't break compat).

By listing I mean having to write this:

<details>
<summary>
get_doc_classes()
</summary>

```python
def get_doc_classes():	
    return [	
        "@VisualScript",	
        "VisualScriptBasicTypeConstant",	
        "VisualScriptBuiltinFunc",	
        "VisualScriptClassConstant",	
        "VisualScriptComment",	
        "VisualScriptComposeArray",	
        "VisualScriptCondition",	
        "VisualScriptConstant",	
        "VisualScriptConstructor",	
        "VisualScriptCustomNode",	
        "VisualScriptDeconstruct",	
        "VisualScriptEditor",	
        "VisualScriptEmitSignal",	
        "VisualScriptEngineSingleton",	
        "VisualScriptExpression",	
        "VisualScriptFunctionCall",	
        "VisualScriptFunctionState",	
        "VisualScriptFunction",	
        "VisualScriptGlobalConstant",	
        "VisualScriptIndexGet",	
        "VisualScriptIndexSet",	
        "VisualScriptInputAction",	
        "VisualScriptIterator",	
        "VisualScriptLists",	
        "VisualScriptLocalVarSet",	
        "VisualScriptLocalVar",	
        "VisualScriptMathConstant",	
        "VisualScriptNode",	
        "VisualScriptOperator",	
        "VisualScriptPreload",	
        "VisualScriptPropertyGet",	
        "VisualScriptPropertySet",	
        "VisualScriptResourcePath",	
        "VisualScriptReturn",	
        "VisualScriptSceneNode",	
        "VisualScriptSceneTree",	
        "VisualScriptSelect",	
        "VisualScriptSelf",	
        "VisualScriptSequence",	
        "VisualScriptSubCall",	
        "VisualScriptSwitch",	
        "VisualScriptTypeCast",	
        "VisualScriptVariableGet",	
        "VisualScriptVariableSet",	
        "VisualScriptWhile",	
        "VisualScript",	
        "VisualScriptYieldSignal",	
        "VisualScriptYield",	
    ]
```

</details>

---

Use glob pattern to fetch all documentation within modules. This removes the need to manually list all classes for which documentation is written in `config.py` via `get_doc_classes()` method, so existing usages are removed from engine's built-in modules.

Removed `get_doc_path()` methods from individual modules as well because all use the same doc path which is `doc_classes`, so the default value is written in `Sconstruct` instead. Modules can still override this path.

Similarly, the default path for module editor icons is provided for consistency with documentation path config mechanism.

The `doc_data_class_path.gen.h` is renamed to `modules_doc_data.gen.h` as it contains path remapping for module's class documentation only.

Tested with `godot --doctool .`

### CODE REVIEW

**Comment by Xrayez** [comment]
This enhancement can be targeted for 4.0, but might be worth including some deprecation/obsolete prints, similarly as done for `can_build()` method signature:

https://github.com/godotengine/godot/blob/aa0fa6b14f62c6aeab04345151cf7c2a9f6df79f/SConstruct#L419-L425

Something along the lines of:
```python
print("Note: get_doc_classes() will become obsolete in 4.0, consider removing it.")
```

But again this change doesn't introduce compatibility breakage in the first place so not sure.

File: editor/SCsub [hunk 1]
```diff
@@ -11,15 +11,17 @@ from compat import open_utf8
 import editor_builders
 
 
-def _make_doc_data_class_path(to_path):
+def _make_modules_doc_data(to_path, doc_pathmap):
```
Comment (akien-mga) [inline editor/SCsub]: It's not only modules though since it includes `doc/classes`?

Comment (Xrayez) [reply editor/SCsub]: If you look at generated header:

<details>
<summary>
doc_data_class_path.gen.h
</summary>

```cpp
static const int _doc_data_class_path_count = 96;
struct _DocDataClassPath { const char* name; const char* path; };
static const _DocDataClassPath _doc_data_class_paths[97] = {
	{"@GDScript", "modules/gdscript/doc_classes"},
	{"@NativeScript", "modules/gdnative/doc_classes"},
	{"@VisualScript", "modules/visual_script/doc_classes"},
	{"ARVRInterfaceGDNative", "modules/gdnative/doc_classes"},
	{"AudioStreamOGGVorbis", "modules/stb_vorbis/doc_classes"},
	{"AudioStreamOpus", "modules/opus/doc_classes"},
	{"BulletPhysicsDirectBodyState", "modules/bullet/doc_classes"},
	{"BulletPhysicsServer", "modules/bullet/doc_classes"},
	{"CSGBox", "modules/csg/doc_classes"},
	{"CSGCombiner", "modules/csg/doc_classes"},
	{"CSGCylinder", "modules/csg/doc_classes"},
	{"CSGMesh", "modules/csg/doc_classes"},
	{"CSGPolygon", "modules/csg/doc_classes"},
	{"CSGPrimitive", "modules/csg/doc_classes"},
	{"CSGShape", "modules/csg/doc_classes"},
	{"CSGSphere", "modules/csg/doc_classes"},
	{"CSGTorus", "modules/csg/doc_classes"},
	{"GDNative", "modules/gdnative/doc_classes"},
	{"GDNativeLibrary", "modules/gdnative/doc_classes"},
	{"GDScript", "modules/gdscript/doc_classes"},
	{"GDScriptFunctionState", "modules/gdscript/doc_classes"},
	{"GDScriptNativeClass", "modules/gdscript/doc_classes"},
	{"GridMap", "modules/gridmap/doc_classes"},
	{"MobileVRInterface", "modules/mobile_vr/doc_classes"},
	{"MultiplayerPeerGDNative", "modules/gdnative/doc_classes"},
	{"NativeScript", "modules/gdnative/doc_classes"},
	{"NetworkedMultiplayerENet", "modules/enet/doc_classes"},
	{"NoiseTexture", "modules/opensimplex/doc_classes"},
	{"OpenSimplexNoise", "modules/opensimplex/doc_classes"},
	{"PacketPeerGDNative", "modules/gdnative/doc_classes"},
	{"PluginScript", "modules/gdnative/doc_classes"},
	{"RegEx", "modules/regex/doc_classes"},
	{"RegExMatch", "modules/regex/doc_classes"},
	{"StreamPeerGDNative", "modules/gdnative/doc_classes"},
	{"UPNP", "modules/upnp/doc_classes"},
	{"UPNPDevice", "modules/upnp/doc_classes"},
	{"VideoStreamGDNative", "modules/gdnative/doc_classes"},
	{"VideoStreamTheora", "modules/theora/doc_classes"},
	{"VideoStreamWebm", "modules/webm/doc_classes"},
	{"VisualScript", "modules/visual_script/doc_classes"},
	{"VisualScriptBasicTypeConstant", "modules/visual_script/doc_classes"},
	{"VisualScriptBuiltinFunc", "modules/visual_script/doc_classes"},
	{"VisualScriptClassConstant", "modules/visual_script/doc_classes"},
	{"VisualScriptComment", "modules/visual_script/doc_classes"},
	{"VisualScriptComposeArray", "modules/visual_script/doc_classes"},
	{"VisualScriptCondition", "modules/visual_script/doc_classes"},
	{"VisualScriptConstant", "modules/visual_script/doc_classes"},
	{"VisualScriptConstructor", "modules/visual_script/doc_classes"},
	{"VisualScriptCustomNode", "modules/visual_script/doc_classes"},
	{"VisualScriptDeconstruct", "modules/visual_script/doc_classes"},
	{"VisualScriptEditor", "modules/visual_script/doc_classes"},
	{"VisualScriptEmitSignal", "modules/visual_script/doc_classes"},
	{"VisualScriptEngineSingleton", "modules/visual_script/doc_classes"},
	{"VisualScriptExpression", "modules/visual_script/doc_classes"},
	{"VisualScriptFunction", "modules/visual_script/doc_classes"},
	{"VisualScriptFunctionCall", "modules/visual_script/doc_classes"},
	{"VisualScriptFunctionState", "modules/visual_script/doc_classes"},
	{"VisualScriptGlobalConstant", "modules/visual_script/doc_classes"},
	{"VisualScriptIndexGet", "modules/visual_script/doc_classes"},
	{"VisualScriptIndexSet", "modules/visual_script/doc_classes"},
	{"VisualScriptInputAction", "modules/visual_script/doc_classes"},
	{"VisualScriptIterator", "modules/visual_script/doc_classes"},
	{"VisualScriptLists", "modules/visual_script/doc_classes"},
	{"VisualScriptLocalVar", "modules/visual_script/doc_classes"},
	{"VisualScriptLocalVarSet", "modules/visual_script/doc_classes"},
	{"VisualScriptMathConstant", "modules/visual_script/doc_classes"},
	{"VisualScriptNode", "modules/visual_script/doc_classes"},
	{"VisualScriptOperator", "modules/visual_script/doc_classes"},
	{"VisualScriptPreload", "modules/visual_script/doc_classes"},
	{"VisualScriptPropertyGet", "modules/visual_script/doc_classes"},
	{"VisualScriptPropertySet", "modules/visual_script/doc_classes"},
	{"VisualScriptResourcePath", "modules/visual_script/doc_classes"},
	{"VisualScriptReturn", "modules/visual_script/doc_classes"},
	{"VisualScriptSceneNode", "modules/visual_script/doc_classes"},
	{"VisualScriptSceneTree", "modules/visual_script/doc_classes"},
	{"VisualScriptSelect", "modules/visual_script/doc_classes"},
	{"VisualScriptSelf", "modules/visual_script/doc_classes"},
	{"VisualScriptSequence", "modules/visual_script/doc_classes"},
	{"VisualScriptSubCall", "modules/visual_script/doc_classes"},
	{"VisualScriptSwitch", "modules/visual_script/doc_classes"},
	{"VisualScriptTypeCast", "modules/visual_script/doc_classes"},
	{"VisualScriptVariableGet", "modules/visual_script/doc_classes"},
	{"VisualScriptVariableSet", "modules/visual_script/doc_classes"},
	{"VisualScriptWhile", "modules/visual_script/doc_classes"},
	{"VisualScriptYield", "modules/visual_script/doc_classes"},
	{"VisualScriptYieldSignal", "modules/visual_script/doc_classes"},
	{"WebRTCDataChannel", "modules/webrtc/doc_classes"},
	{"WebRTCDataChannelGDNative", "modules/gdnative/doc_classes"},
	{"WebRTCMultiplayer", "modules/webrtc/doc_classes"},
	{"WebRTCPeerConnection", "modules/webrtc/doc_classes"},
	{"WebRTCPeerConnectionGDNative", "modules/gdnative/doc_classes"},
	{"WebSocketClient", "modules/websocket/doc_classes"},
	{"WebSocketMultiplayerPeer", "modules/websocket/doc_classes"},
	{"WebSocketPeer", "modules/websocket/doc_classes"},
	{"WebSocketServer", "modules/websocket/doc_classes"},
	{NULL, NULL}
};
```
</details>

You see how those include module doc paths only. Looks like it's used for doc dump for classes with non-default documentation path. I'm not sure whether it makes sense to generalize the usage of this header to include other non-core stuff.

So the process of generating *doc data* and *doc paths* are different here. `_make_modules_doc_data` could be renamed further to `_make_modules_doc_paths` to be more specific.

**Comment by Xrayez** [comment]
Upon further testing I realized that this only works for doc classes which were already generated per module (meaning if you delete the doc classes from a module, then a module's doc paths are not even generated within the header (I made a silly assumption).

What's needed is a better differentiation between core/module classes on the engine level, and ability to get a list of module-only classes provided by `ClassDB` methods or similar at run-time...

`get_doc_path` can still be made optional though, with the default path pointing to `modules/x/doc_classes`.

I'll just close this PR to prevent from merging now, have to think of better ways to solve this.

**Pull request closed without merging**